Source/CarthageKit/Project.swift

```diff
@@ -675,19 +681,46 @@ public final class Project {
 				return self.buildOrderForResolvedCartfile(resolvedCartfile, dependenciesToInclude: dependenciesToBuild)
 			}
 			.flatMap(.Concat) { dependency -> SignalProducer<BuildSchemeProducer, CarthageError> in
-				let dependencyPath = self.directoryURL.URLByAppendingPathComponent(dependency.project.relativePath, isDirectory: true).path!
+				let project = dependency.project
+				let version = dependency.version.commitish
+
+				let dependencyPath = self.directoryURL.URLByAppendingPathComponent(project.relativePath, isDirectory: true).path!
 				if !NSFileManager.defaultManager().fileExistsAtPath(dependencyPath) {
 					return .empty
 				}
 
-				return buildDependencyProject(dependency.project, self.directoryURL, withOptions: options, sdkFilter: sdkFilter)
+				var options = options
+				let cleanDerivedDataIfNeeded: SignalProducer<(), CarthageError>
+
+				if options.derivedDataPath != nil {
+					cleanDerivedDataIfNeeded = .empty
+				} else {
```

Comment:
I was thinking that we should opt out the behavior when the `--derived-data` option is specified, but may not need to do that.